java/vector/src/main/java/org/apache/arrow/vector/util/Validator.java

```diff
@@ -33,101 +38,140 @@
  */
 public class Validator {
 
-    /**
-     * Validate two arrow schemas are equal.
-     *
-     * @param schema1 the 1st shema to compare
-     * @param schema2 the 2nd shema to compare
-     * @throws IllegalArgumentException if they are different.
-     */
-    public static void compareSchemas(Schema schema1, Schema schema2) {
-        if (!schema2.equals(schema1)) {
-            throw new IllegalArgumentException("Different schemas:\n" + schema2 + "\n" + schema1);
-        }
+  /**
+   * Validate two arrow schemas are equal.
+   *
+   * @param schema1 the 1st shema to compare
+   * @param schema2 the 2nd shema to compare
+   * @throws IllegalArgumentException if they are different.
+   */
+  public static void compareSchemas(Schema schema1, Schema schema2) {
```

Comment:
if most of these methods haven't changed, and only formatting was fixed, it might be good to revert them to avoid confusion. not sure what the project standards are in that regards...